WordPress/gutenberg
# Navigation Block: Fix warning error when no ref

Related: #47453

## What?
This PR fixes a problem that caused a warning error when rendering navigation blocks.

![error](https://user-images.githubusercontent.com/54422211/215011760-326917d5-c022-4b43-9500-16befdb6b79f.png)

## Why?
My understanding is that if any changes are made to the navigation block, the content is stored in the database and `ref` attribute is given. It then retrieves the post from the database based on this ref, stores it in the `$navigation_post` variable, and checks whether the navigation block is nested based on its content.

However, if the user has not made any changes to this block, the ref does not exist. As a result, `$navigation_post` is undefined in the current implementation and an error occurs.

## How?
~I have moved the process of checking for nesting into an if statement that is only executed if the block has a ref.~

Used `$inner_blocks` instead of $navigation_post in the `block_core_navigation_block_contains_core_navigation` argument to prevent undefined errors. The `block_core_navigation_get_menu_items_at_location` function has also been adjusted to reflect the change in argument types.

## Testing Instructions

- Activate Twenty Twenty Three theme.
- Clear all customizations made on Template and Template Part.
- Go to `http://localhost:8888/wp-admin/edit.php?post_type=wp_navigation` and remove all navigation.
- Access the top page of the site.
- Confirm that no PHP errors occur.

### Regression Test

Please confirm that block_core_navigation_get_menu_items_at_location is working properly as before by following the steps below.

- Remove all your `wp_navigation` posts and Classic Menus from your site
- Create a brand new Navigation using the Navigation block. Be sure to save it.
- Now open your MySQL client and update the `post_content` of your `wp_navigation` post with the following content, being sure to replace `%%REPLACE_ME%%` with the ID of the `wp_navigation` post _**YOU**_ just created:
```

```
- Reload the same Post.
- See that it loads ok and the Navigation block is not rendered (because there are no blocks).
